subsys/net/ip/ipv6.c

```diff
@@ -2149,6 +2149,9 @@ int net_ipv6_send_ns(struct net_if *iface,
 		if (!nbr) {
 			NET_DBG("Could not create new neighbor %s",
 				net_sprint_ipv6_addr(&ns_hdr->tgt));
+			if (pending) {
+				net_pkt_unref(pending);
+			}
 			goto drop;
```

Comment:
Looks good but one nit pick, please put empty line before the goto.

Reply:
OK, I just pushed updated patch